src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -75,6 +75,32 @@ static std::atomic<uint64_t> tx_hashes_cached_count(0);
 static std::atomic<uint64_t> block_hashes_calculated_count(0);
 static std::atomic<uint64_t> block_hashes_cached_count(0);
 
+#define CHECK_AND_ASSERT_THROW_MES_L1(expr, message) {if(!(expr)) {MWARNING(message); throw std::runtime_error(message);}}
+
+namespace
+{
+  static inline unsigned char *operator &(ec_point &point) {
+    return &reinterpret_cast<unsigned char &>(point);
+  }
+  static inline const unsigned char *operator &(const ec_point &point) {
+    return &reinterpret_cast<const unsigned char &>(point);
+  }
```

Comment:
I don't know why, but these local operator overloads seem to confuse clang compiler on OSX possibly due to some template serialization gotcha. Removing these and instead using `reinterpret_cast<unsigned char*>(&A)` etc for `ge_frombytes_vartime` and `ge_p3_tobytes` resolved this issue.

Reply:
Can you paste the error ? This exists in src/crypto already, and it apparently works there.

Reply:
Here's the log: https://paste.debian.net/995831/
Compiler version:
```
-- The C compiler identification is AppleClang 9.0.0.9000038
-- The CXX compiler identification is AppleClang 9.0.0.9000038
```